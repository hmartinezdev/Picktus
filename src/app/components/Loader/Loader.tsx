import Photo from '@assets/svg/pix.svg';
import colors from '@constants/colors';
import React, { Component } from 'react';

class Loader extends Component {
  public render(): React.ReactElement<Loader> {
    return (
      <div className="container">
        <Photo className="photo photo--first" />
        <Photo className="photo photo--second" />
        <style jsx>{`
          .container {
            position: relative;
            width: 100%;
            height: 100%;
          }

          .photo {
            max-width: 100%;
            max-height: 100%;
            position: absolute;
          }

          .photo--second {
            animation: photo 3000ms ease-out infinite;
          }

          .photo--second .check {
            stroke-dasharray: 1000;
            stroke-dashoffset: 1000;
            stroke-width: 22;
            fill: none;
            stroke-meterlimit: 10;
            animation: check 3000ms ease-out infinite;
          }

          @keyframes photo {
            0% {
              transform: translateY(-100%);
              opacity: 0;
            }

            20% {
              opacity: 1;
            }

            40% {
              transform: translateY(0%);
              opacity: 1;
            }

            80%,
            100% {
              opacity: 1;
            }
          }

          @keyframes check {
            0% {
              stroke-dashoffset: 1000;
            }

            40% {
              stroke-dashoffset: 1000;
              fill: ${colors.green};
            }

            80%,
            100% {
              fill: ${colors.lightGreen};
              stroke-dashoffset: 0;
            }
          }
        `}</style>
      </div>
    );
  }
}

export default Loader;
