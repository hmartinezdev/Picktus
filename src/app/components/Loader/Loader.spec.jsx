import { shallow } from 'enzyme';
import React from 'react';
import Loader from './Loader';

let wrapper;

const setup = () => shallow(<Loader />);

describe('<Loader />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
