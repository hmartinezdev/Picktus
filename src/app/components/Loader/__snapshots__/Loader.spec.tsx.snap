// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render properly 1`] = `
<div
  className="container"
>
  <Component
    className="photo photo--first"
  />
  <Component
    className="photo photo--second"
  />
  <style
    jsx={true}
  >
    
          .container {
            position: relative;
            width: 100%;
            height: 100%;
          }

          :global(.photo) {
            max-width: 100%;
            max-height: 100%;
            position: absolute;
          }

          :global(.photo--second) {
            animation: photo 3000ms ease-out infinite;
          }

          :global(.photo--second .check) {
            stroke-dasharray: 1000;
            stroke-dashoffset: 1000;
            stroke-width: 22;
            fill: none;
            stroke-meterlimit: 10;
            animation: check 3000ms ease-out infinite;
          }

          @keyframes photo {
            0% {
              transform: translateY(-100%);
              opacity: 0;
            }

            20% {
              opacity: 1;
            }

            40% {
              transform: translateY(0%);
              opacity: 1;
            }

            80%,
            100% {
              opacity: 1;
            }
          }

          @keyframes check {
            0% {
              stroke-dashoffset: 1000;
            }

            40% {
              stroke-dashoffset: 1000;
              fill: #75ab88;
            }

            80%,
            100% {
              fill: #b3e099;
              stroke-dashoffset: 0;
            }
          }
        
  </style>
</div>
`;
